### PR TITLE
Transfer: add API to retrieve transfer source locations

### PR DESCRIPTION
**Pull request by mistydemeo**

### CODE REVIEW

File: src/dashboard/src/components/transfer/views.py [hunk 1]
```diff
@@ -64,6 +64,16 @@ def grid(request):
     hide_features = helpers.hidden_features()
     return render(request, 'transfer/grid.html', locals())
 
+def transfer_source_locations(request):
+    try:
+        return helpers.json_response(storage_service.get_location(purpose="TS"))
+    except:
```
Comment (Hwesta) [inline src/dashboard/src/components/transfer/views.py:69]: Log something about why it failed? logger.exception?

**Comment by Hwesta** [comment]
What's the use case for this? Since I don't see anything that calls it yet (did you forget to add?).

**Comment by mistydemeo** [comment]
It's being used by the new transfer browser. The current version interpolates the transfer source directories into JavaScript at rendertime using Django templates; the new transfer browser instead requests them clientside after the page loads. Aside from being cleaner, it'll also allow the user to refresh the source directories after the page has been loaded without reloading the entire page (for example, if the storage service was broken when the page was loaded).

**Comment by Hwesta** [comment]
Ah, nice!

Possibly out of scope for this, but it seems like this endpoint and `children/location/<location_uuid>/` from filesystem_ajax/urls.py should be together.

:+1:

File: src/dashboard/src/components/transfer/views.py (hunk 1 above)
Comment (mistydemeo) [inline src/dashboard/src/components/transfer/views.py:69]: Added a `logger.exception` call.

**Comment by mistydemeo** [comment]
> Possibly out of scope for this, but it seems like this endpoint and children/location/<location_uuid>/ from filesystem_ajax/urls.py should be together.

Yeah, the current placement is a bit strange. I think the `filesystem_ajax` one is there because it was placed with the original version which actually hit the filesystem. Those APIs didn't really get reorganized when this got moved to the storage service.

**Comment by Hwesta** [comment]
I think another PR for the API cleanup stuff, then. Probably related to the transfer browser changes.

**Pull request merged**